sharding-core/sharding-core-merge/src/main/java/org/apache/shardingsphere/core/merge/dal/show/ShowTablesMergedResult.java

```diff
@@ -17,42 +17,43 @@
 
 package org.apache.shardingsphere.core.merge.dal.show;
 
-import org.apache.shardingsphere.core.constant.ShardingConstant;
-import org.apache.shardingsphere.core.execute.sql.execute.result.QueryResult;
-import org.apache.shardingsphere.core.metadata.table.TableMetas;
-import org.apache.shardingsphere.core.rule.ShardingRule;
+import lombok.RequiredArgsConstructor;
 
-import java.sql.SQLException;
-import java.util.HashMap;
+import java.util.Collections;
 import java.util.List;
-import java.util.Map;
 
 /**
- * Merged result for show tables.
+ * Merged result for show tableNames.
  *
  * @author zhangliang
  * @author panjuan
+ * @author sunbufu
  */
-public final class ShowTablesMergedResult extends LogicTablesMergedResult {
+@RequiredArgsConstructor
+public final class ShowTablesMergedResult extends LocalMergedResultAdapter {
     
-    private static final Map<String, Integer> LABEL_AND_INDEX_MAP = new HashMap<>(1, 1);
+    private final List<String> tableNames;
```

Comment:
What is the different of tableNames & tableTypes?
Should they split to two class to handler different SQL?